custom_components/fordpass/sensor.py

```diff
@@ -67,28 +67,28 @@ def __init__(self, coordinator, sensor, options):
         self._attr = {}
         self.coordinator = coordinator
         self.units = coordinator.hass.config.units
-        self.data = coordinator.data["metrics"]
-        self.events = coordinator.data["events"]
-        self.states = coordinator.data["states"]
+        self.data = coordinator.data.get("metrics", {})
+        self.events = coordinator.data.get("events", {})
+        self.states = coordinator.data.get("states", {})
         self._device_id = "fordpass_" + sensor
         # Required for HA 2022.7
         self.coordinator_context = object()
 
     def get_value(self, ftype):
         """Get sensor value and attributes from coordinator data"""
-        self.data = self.coordinator.data["metrics"]
-        self.events = self.coordinator.data["events"]
-        self.states = self.coordinator.data["states"]
+        self.data = self.coordinator.data.get("metrics", {})
+        self.events = self.coordinator.data.get("events", {})
+        self.states = self.coordinator.data.get("states", {})
         self.units = self.coordinator.hass.config.units
         if ftype == "state":
             if self.sensor == "odometer":
                 return self.data.get("odometer", {}).get("value")
                     #return self.data.get("odometer", {}).get("value", {})
             if self.sensor == "fuel":
-                fuel_level = self.data.get("fuelLevel", {}).get("value", 0)
+                fuel_level = self.data.get("fuelLevel", {}).get("value")
                 if fuel_level is not None:
                     return round(fuel_level)
-                battery_soc = self.data.get("xevBatteryStateOfCharge", {}).get("value", 0)
+                battery_soc = self.data.get("xevBatteryStateOfCharge", {}).get("value")
```

Comment:
Removed the default returning of "0" from here and line 88.
Because it was 0, for EV's it would display fuelLevel 0.